greenheart/simulation/greenheart_simulation.py

```diff
@@ -33,6 +40,68 @@
 
 pd.options.mode.chained_assignment = None  # default='warn'
 
+# """
+# Recursively converts complex types to JSON/YAML-compatible formats.
+# Handles:
+# - `np.ndarray` -> list
+# - `tuple` -> list
+# - `np.generic` (e.g., `np.float64`, `np.int32`) -> native Python types
+# - `pandas.DataFrame` -> list of dicts, recursively processed
+# - `pandas.Series` -> list, recursively processed
+# - `attrs` objects -> dict of serialized attributes
+# - Handles deeply nested structures
+
+# Note: this function was originally created by ChatGPT and edited manually to work as desired
+# """
```

Comment:
Do you mean to retain this commented out docstring here?

Reply:
Nope, I've removed it